phac-nml/irida-next
# Project Attachments: Index, Table Component, Upload

## What does this PR do and why?
This PR adds the UI for project attachments. This includes adding a `table_component` for attachments, and upload functionality for project attachments. Ransack sort and filter, pagination, and per page limit are also implemented.

To be addressed in future PRs:
1. Add delete action functionality
2. Add Group attachments

## Screenshots or screen recordings
![image](https://github.com/user-attachments/assets/3cae6058-ff96-46d6-b08d-ae36815b089d)

## How to set up and validate locally
1. Navigate to a project
3. Click "Files" on the sidebar
4. Verify `No Files` state
5. Verify the upload works by uploading a bunch of files, preferably with different formats and types
6. Verify per page selection, pagination, and sorting/filtering works as expected
7. Click on a filename and ensure it downloads

Other tests:
8. Verify project `guests` cannot see the `Files` sidebar link in projects
9. Verify project `analysts` cannot see the `Upload Files` button on the project attachments index page
10. After uploading a file, wait ~1 minute to verify the table automatically updates the `Uploaded time_ago`
11. Verify sort, filter, and limit selection work as expected, and work together (ie: sort, then add a filter, then select a limit; all should still be enabled)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
